eng/packages/http-client-csharp/generator/TestProjects/Local/Mgmt-TypeSpec/src/Generated/LongRunningOperation/FooOperationSource.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.ClientModel.Primitives;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager;
+using MgmtTypeSpec.Models;
+
+namespace MgmtTypeSpec
+{
+    /// <summary></summary>
+    internal partial class FooOperationSource : IOperationSource<FooResource>
+    {
+        private readonly ArmClient _client;
+
+        internal FooOperationSource(ArmClient client)
+        {
+            _client = client;
+        }
+
+        FooResource IOperationSource<FooResource>.CreateResult(Response response, CancellationToken cancellationToken)
+        {
+            using JsonDocument document = JsonDocument.Parse(response.ContentStream);
+            FooData data = FooData.DeserializeFooData(document.RootElement, new ModelReaderWriterOptions("W"));
+            return new FooResource(_client, data);
+        }
+
+        async ValueTask<FooResource> IOperationSource<FooResource>.CreateResultAsync(Response response, CancellationToken cancellationToken)
+        {
+            using JsonDocument document = await JsonDocument.ParseAsync(response.ContentStream, default, cancellationToken).ConfigureAwait(false);
+            FooData data = FooData.DeserializeFooData(document.RootElement, new ModelReaderWriterOptions("W"));
```

Comment:
`new ModelReaderWriterOptions("W")` is needed here because of options is not optional in serialization
https://github.com/Azure/azure-sdk-for-net/blob/44a1cc564d07a8f964f29a77581e8849861b62d2/eng/packages/http-client-csharp/generator/TestProjects/Local/Mgmt-TypeSpec/src/Generated/FooData.Serialization.cs#L69

Reply:
I think there is a static instance for this in one of our extension classes maybe we should use that instead of instantiating a new every time we get in this method.

Reply:
here it is: https://github.com/live1206/azure-sdk-for-net/blob/c5ca437d011c352d5bf0c5103857efb071a93927/eng/packages/http-client-csharp/generator/TestProjects/Local/Mgmt-TypeSpec/src/Generated/Internal/ModelSerializationExtensions.cs#L19
I wonder how we could use it.
In MrwSerializationDefinition, they are using this - but since we do not really have the reference to that typeprovider here, I wonder if there could be a way that we could do it without hard coding its name here.

Reply:
I noticed that before, but did not think it's necessary to reuse it.

Updated to use it in the generated code.